src/cplus_plugin/gui/qgis_cplus_main.py

```diff
@@ -1983,3 +1996,8 @@ def report_job_is_for_current_scenario(self, scenario_id: str) -> bool:
             return True
 
         return False
+
+    def on_progress_dialog_cancelled(self):
```

Comment:
@gkahiu We should re-enable the `run_scenario_btn` button after the scenario analysis has finished, not after user has canceled/closed the progress dialog.

Reply:
Thanks @Samweli. 

There are three scenarios where we re-enable the button:
1. When reporting has finished (after scenario analysis), see [here](https://github.com/kartoza/cplus-plugin/pull/223/files#diff-7c0d242d9c04008b41931ad20570d5701ac4f0789738aa195107c012e349a17bR1971);

2. When the user has explicitly cancelled the analysis, to allow them to re-run the analysis again;

3. When an error occurs.

I hope this helps to clarify.